source/core/ut_suite_manager.pkb

```diff
@@ -327,5 +339,166 @@ create or replace package body ut_suite_manager is
+  procedure validate_paths(a_paths in ut_varchar2_list) is
+    l_path varchar2(32767);
+  begin
+    if a_paths is null or a_paths.count = 0 then
+      raise_application_error(ut_utils.gc_path_list_is_empty, 'Path list is empty');
+    else
+      for i in 1 .. a_paths.count loop
+        l_path := a_paths(i);
+        if regexp_like(l_path, '^\w+(\.\w+){0,2}$') or regexp_like(l_path, '^\w+:\w+(\.\w+)*$') then
+          null;
+        else
+          raise_application_error(ut_utils.gc_invalid_path_format, 'Invalid path format: ' || nvl(l_path, 'NULL'));
+        end if;
+      end loop;
+    end if;
+  end validate_paths;
+
+  procedure run(a_paths in ut_varchar2_list, a_reporter in ut_reporter) is
+    l_paths           ut_varchar2_list;
+    l_path            varchar2(32767);
+    l_schema          varchar2(4000);
+    l_objects_to_run  ut_objects_list := ut_objects_list();
+    l_schema_suites   tt_schema_suits;
+    l_index           varchar2(4000 char);
+    l_suite           ut_test_suite;
+    l_suite_path      varchar2(4000);
+    l_root_suite_name varchar2(4000);
+    l_reporter        ut_reporter := a_reporter;
+  
+    function clean_paths(a_paths ut_varchar2_list) return ut_varchar2_list is
+      l_paths ut_varchar2_list := ut_varchar2_list();
+    begin
+      l_paths.extend(a_paths.count);
+      for i in 1 .. a_paths.count loop
+        l_paths(i) := trim(lower(a_paths(i)));
+      end loop;
+      l_paths := set(l_paths);
+      return l_paths;
+    end clean_paths;
+  
+    procedure set_skipped_flag(a_suite in out nocopy ut_test_object, a_path varchar2) is
+      l_root      constant varchar2(32767) := regexp_substr(a_path, '\w+');
+      l_rest_path constant varchar2(32767) := regexp_substr(a_path, '\.(.+)', subexpression => 1);
+      l_item  ut_test_object;
+      l_items ut_objects_list := ut_objects_list();
+      l_object_name varchar2(32767);
+      
+    begin
+      a_suite.set_ignore_flag(false);
+    
+      if a_path is not null then
+      
+        for i in 1 .. a_suite.items.count loop
+        
+          l_item := treat(a_suite.items(i) as ut_test_object);
+        
+          l_object_name := l_item.object_name;
+          --l_object_name := regexp_substr(l_object_name,'\w+$'); -- temporary fix. seems like suite have suitepath in object_name
+          if regexp_like(l_object_name, l_root, modifier => 'i') then
+            
+            set_skipped_flag(l_item, l_rest_path);          
+            l_items.extend;
+            l_items(l_items.count) := l_item;
+
+          end if;
+          
+        end loop;
+        
+        a_suite.items := l_items;
+      
+        if l_items.count = 0 then--not l_found then
+          raise_application_error(-20203, 'Suite note found');
+        end if;
+      end if;
+    end set_skipped_flag;
+  begin
+    l_paths := clean_paths(a_paths);
+  
+    validate_paths(l_paths);
+  
+    -- current implementation operates only on a single path
```

Comment:
Maybe it would be better to extract the find/filter logic into separate procedure?